lib/portlayer/exec/exec.go

```diff
@@ -155,18 +159,123 @@ func eventCallback(ie events.Event) {
 						container.onStop()
 					}
 					log.Debugf("Container(%s) state set to %s via event activity",
-						container.ExecConfig.ID, newState.String())
+						container, newState.String())
 					// regardless of update success failure publish the container event
 					publishContainerEvent(container.ExecConfig.ID, ie.Created(), ie.String())
 				}()
 			case StateRemoved:
-				log.Debugf("Container(%s) %s via event activity", container.ExecConfig.ID, newState.String())
+				log.Debugf("Container(%s) %s via event activity", container, newState.String())
 				Containers.Remove(container.ExecConfig.ID)
 				publishContainerEvent(container.ExecConfig.ID, ie.Created(), ie.String())
 			}
+		} else {
+			switch ie.String() {
+			case events.ContainerRelocated:
+				log.Debugf("Container %s received ContainerRelocated", container)
+
+				// container state has changed so we need to update the container attributes
+				// we'll do this in a go routine to avoid blocking
+				go func() {
+					ctx, cancel := context.WithTimeout(context.Background(), propertyCollectorTimeout)
+					defer cancel()
+
+					err := container.Refresh(ctx)
+					if err != nil {
+						log.Errorf("Event driven container update failed for %s with %s", container, err.Error())
+					}
+				}()
+			}
 		}
+		return
+	}
+}
+
+// listens migrated events and connects the file backed serial ports
+func hostEventCallback(ctx context.Context, ie events.Event) {
+	defer trace.End(trace.Begin(""))
+
+	switch ie.String() {
+	// https://www.vmware.com/support/developer/vc-sdk/visdk25pubs/ReferenceGuide/vim.event.EnteringMaintenanceModeEvent.html
+	// This event records that a host has begun the process of entering maintenance mode. All virtual machine operations are blocked, except the following:
+	// MigrateVM
+	// PowerOffVM
+	// SuspendVM
+	// ShutdownGuest
+	// StandbyGuest
+	//
+	// Because of that limitation the only thing that we can do is shutting down the guest without calling reconfigure
+	case events.HostEnteringMaintenanceMode:
+		log.Debugf("Received %s event", ie)
+		ref := ie.Reference()
+
+		// we are interested with running vms
+		state := new(State)
```

Comment:
State is an int type. Do we really need a pointer for it? It makes code looking ugly. Of course, it is not a scope of this PR.